### PR TITLE
Update licensing

### PR DESCRIPTION
**Pull request by timbussmann**
relates to https://github.com/Particular/PlatformDevelopment/issues/1386
update the licensing package to make ServiceControl able to read licenses from the new file based locations and to import licenses into the filesystem as well.

these changes are untested

### CODE REVIEW

File: src/ServiceControl/packages.config [hunk 1]
```diff
@@ -27,7 +27,7 @@
   <package id="NServiceBus.NLog" version="1.1.0" targetFramework="net452" />
   <package id="Owin" version="1.0" targetFramework="net452" />
   <package id="Owin.Metrics" version="0.3.7" targetFramework="net452" />
-  <package id="Particular.Licensing.Sources" version="0.1.59" targetFramework="net452" />
+  <package id="Particular.Licensing.Sources" version="1.0.0-alpha0004" targetFramework="net452" />
```
Comment (WilliamBZA) [inline src/ServiceControl/packages.config]: What is the expected timeframe for a non-alpha?

File: src/ServiceControlInstaller.Engine/LicenseMgmt/LicenseManager.cs [hunk 2]
```diff
@@ -40,17 +41,16 @@ public static bool TryImportLicense(string licenseFile, out string errorMessage)
...
-                }
+                new RegistryLicenseStore(Registry.LocalMachine).StoreLicense(licenseText);
             }
             catch (Exception)
             {
                 errorMessage = "Failed to import license into the registry";
                 return false;
             }
 
+            //TODO: import license into new registry location
```
Comment (WilliamBZA) [inline src/ServiceControlInstaller.Engine/LicenseMgmt/LicenseManager.cs]: Todo when?

Comment (timbussmann) [inline src/ServiceControlInstaller.Engine/LicenseMgmt/LicenseManager.cs]: when the new licensing package is available (this is why this PR is marked as WIP)

**Comment by timbussmann** [comment]
ready for review. Please keep in mind these changes are untested and are a proposal on how to deal with the new licensing locations in SC.

**Comment by timbussmann** [comment]
fyi: I'm going on vacation, so I'll not update this PR. Please feel free to hijack this PR.

**Comment by bording** [comment]
I've rebased this, so it should be good to go.

@WilliamBZA To answer your question about timeframe for non-alpha, now that .NET Core is out, we should be able to release 1.0.0 soon.

**Comment by WilliamBZA** [comment]
Thanks @timbussmann and @bording.

Since this repo doesn't have a develop branch we'll have to keep this PR open until the license library is released so we don't release with an alpha or beta version.

I will rebase and upgrade this once 1.0.0 is released.

**Comment by bording** [comment]
@WilliamBZA A shiny new 1.0.0 release is available now, so I've rebased this once again!

**Comment by bording** [comment]
@WilliamBZA @Particular/servicecontrol-maintainers Any reason this hasn't been merged yet? It would be good to get a version out that knows how to write the license file in the correct place for NSB7!

**Pull request merged**